slatedb/src/compactor.rs

```diff
@@ -373,16 +377,41 @@ impl CompactorEventHandler {
     ) -> Result<Self, SlateDBError> {
         let stored_manifest =
             StoredManifest::load(manifest_store.clone(), system_clock.clone()).await?;
+        let manifest_compactor_epoch = stored_manifest.manifest().compactor_epoch;
         let manifest = FenceableManifest::init_compactor(
```

Comment:
The protocol here is that we:

1. Init the manifest and make sure we own `compactor_epoch` in the `.manifest` file.
2. We then init the compaction file (line 400 below) to make sure we own `compactor_epoch` in the `.compactions` file.
3. We then proceed to update the `.compactions` file as compactions are added/removed to the compactor state.
4. `.manifest` polling should detect `compactor_epoch` fencing in the manifest file.
5. `.compactions` writes should detect `compactor_epoch` fencing in the compactions file.